core/templates/dev/head/pages/exploration-player-page/services/audio-translation-language.service.spec.ts

```diff
@@ -25,72 +25,206 @@ describe('Audio translation language service', () => {
     atls = TestBed.get(AudioTranslationLanguageService);
+
+    spyOn(window.speechSynthesis, 'getVoices').and.returnValue([{
+      'default': false,
+      lang: 'en-GB',
+      localService: false,
+      name: 'UK English',
+      voiceURI: 'UK English'
+    }]);
   });
 
   it('should properly initialize the current audio language when ' +
      'a preferred language is set', () => {
     let allAudioLanguageCodesInExploration = ['hi-en', 'en'];
     let preferredLanguageCode = 'hi-en';
     let explorationLanguageCode = 'hi';
+    const automaticTextToSpeechEnabled = false;
+
     atls.init(allAudioLanguageCodesInExploration, preferredLanguageCode,
-      explorationLanguageCode, false);
+      explorationLanguageCode, automaticTextToSpeechEnabled);
+    expect(atls.isAutomaticTextToSpeechEnabled()).toBe(
+      automaticTextToSpeechEnabled);
     expect(atls.getCurrentAudioLanguageCode()).toEqual('hi-en');
+    expect(atls.getallAudioLanguageCodesInExploration()).toBe(
+      allAudioLanguageCodesInExploration);
     atls.clearCurrentAudioLanguageCode();
 
     allAudioLanguageCodesInExploration = ['hi-en', 'en'];
     preferredLanguageCode = 'en';
     explorationLanguageCode = 'hi';
+
     atls.init(allAudioLanguageCodesInExploration, preferredLanguageCode,
-      explorationLanguageCode, false);
+      explorationLanguageCode, automaticTextToSpeechEnabled);
     expect(atls.getCurrentAudioLanguageCode()).toEqual('en');
     atls.clearCurrentAudioLanguageCode();
 
     allAudioLanguageCodesInExploration = ['hi-en'];
     preferredLanguageCode = 'en';
     explorationLanguageCode = 'hi';
+
     atls.init(allAudioLanguageCodesInExploration, preferredLanguageCode,
       explorationLanguageCode, false);
     expect(atls.getCurrentAudioLanguageCode()).toEqual('hi-en');
+    expect(atls.getallAudioLanguageCodesInExploration()).toBe(
+      allAudioLanguageCodesInExploration);
   });
 
   it('should initialize the current audio language when ' +
      'no preferred language is set and the exploration contains an audio ' +
      'language that is related to the exploration language', () => {
-    let allAudioLanguageCodesInExploration = ['hi-en', 'en'];
-    let preferredLanguageCode = null;
-    let explorationLanguageCode = 'hi';
+    const allAudioLanguageCodesInExploration = ['hi-en', 'en'];
+    const preferredLanguageCode = null;
+    const explorationLanguageCode = 'hi';
     atls.init(allAudioLanguageCodesInExploration, preferredLanguageCode,
       explorationLanguageCode, false);
     expect(atls.getCurrentAudioLanguageCode()).toEqual('hi-en');
+    expect(atls.getallAudioLanguageCodesInExploration()).toBe(
+      allAudioLanguageCodesInExploration);
+  });
+
+  it('should initialize the current audio language when ' +
+    'no preferred language is set and the exploration contains an audio ' +
+    'language that is no related to the exploration language', () => {
+    const allAudioLanguageCodesInExploration = ['hi', 'en'];
```

Comment:
Would it make sense to define these as global? seems to be used in multiple places...

Reply:
Done.